lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -854,19 +856,91 @@ SwiftRuntimeTypeVisitor::VisitImpl(std::optional<unsigned> visit_only,
     return success;
   }
 
-  // FIXME: Remove this entire mode.
-  assert(!m_include_clang_types || (m_include_clang_types && count_only));
-  if (m_include_clang_types && count_only) {
+  auto visit_clang_type =
+      [&](CompilerType clang_type) -> llvm::Expected<unsigned> {
+    auto swiftify = [&](CompilerType type) {
+      if (!type.GetTypeSystem().isa_and_nonnull<TypeSystemClang>())
+        return type;
+      CompilerType swift_type = ts.ConvertClangTypeToSwiftType(type);
+      return swift_type ?: type;
+    };
+    unsigned depth = 0;
+    auto n_or_err =
+        clang_type.GetNumChildren(m_omit_empty_base_classes, &m_exe_ctx);
+    if (!n_or_err)
+      return n_or_err.takeError();
+    unsigned n = *n_or_err;
+    bool is_signed, is_enum = clang_type.IsEnumerationType(is_signed);
+    if (count_only)
+      return is_enum ? 1 : n;
+    if (is_enum) {
+      auto get_name = [&]() -> std::string { return "rawValue"; };
+      auto get_info = [&]() -> llvm::Expected<ChildInfo> {
+        auto ts =
+            clang_type.GetTypeSystem().dyn_cast_or_null<TypeSystemClang>();
+
+        if (!ts)
+          return llvm::createStringError("no clang type system");
+        clang::EnumDecl *enum_decl = ts->GetAsEnumDecl(clang_type);
+        if (!enum_decl)
+          return llvm::createStringError("no enum decl");
+        swift::Demangle::Demangler dem;
+        CompilerType raw_value =
+            CompilerType(ts, enum_decl->getIntegerType().getAsOpaquePtr());
+
+        auto bit_size =
+            raw_value.GetBitSize(m_exe_ctx.GetBestExecutionContextScope());
+        if (!bit_size)
+          return bit_size.takeError();
+        ChildInfo child;
+        child.byte_size = *bit_size / 8;
+        child.byte_offset = 0;
```

Comment:
Is `byte_offset = 0`  really correct here? If this is a struct with an objc field won't this cause problems?